packages/visibility_detector/test/render_visibility_detector_test.dart

```diff
@@ -134,4 +134,21 @@ void main() {
 
     expect(detector.debugScheduleUpdateCount, 0);
   });
+
+  testWidgets('RVS can schedule an update for a RO that is not laid out',
+      (WidgetTester tester) async {
+    final RenderVisibilityDetector detector = RenderVisibilityDetector(
+      key: Key('test'),
+      onVisibilityChanged: (_) { fail('should not get called'); },
+    );
+
+    // Force an out of band update to get scheduled without laying out.
+    detector.onVisibilityChanged = (_) {
```

Comment:
without the changes in this patch, this causes an exception because the box implementation tries to access `semanticBounds` without having a `size`.